File: leet-code/455. Assign Cookies/AssignCookies455_2.java
package com.example.algorithm;

import java.util.Arrays;

public class AssignCookies455_2 {

    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int child = 0, cookie = 0;
        while (child < g.length && cookie < s.length) {
            if (g[child] <= s[cookie]) child++;
            cookie++;
        }
        return child;
    }
}

File: leet-code/455. Assign Cookies/AssignCookies455_2.kt
package com.example

class AssignCookies455_2 {
    fun findContentChildren(g: IntArray, s: IntArray): Int {
        g.sort()
        s.sort()
        var contentSum = 0
        var gIdx = 0
        for (cookie in s) {
            if (gIdx == g.size) {
                break
            }
            if (g[gIdx] > cookie) {
                continue
            }
            contentSum += 1
            gIdx += 1
        }
        return contentSum
    }
}
